examples/tensorrt_llm/common/base_engine.py

```diff
@@ -295,30 +305,54 @@ async def publish_kv_cache_events_task(self):
             return
 
         events = self._llm_engine.get_kv_cache_events_async(timeout=5)
-        async for event_list in events:
-            for event in event_list:
-                data = event["data"]
-                if data["type"] == "stored":
-                    parent_hash = data["parent_hash"]
-                    for block in data["blocks"]:
-                        tokens = []
-                        for token in block["tokens"]:
-                            tokens.append(int(token["token_id"]))
-
-                        # Note: Currently data does not have lora_id.
-                        # Using 0 as default value. If later data has
-                        # lora_id, we need to verify if this is correct.
-                        lora_id = data.get("lora_id", 0)
-                        self._kv_cache_events_publisher.stored_event(
-                            parent_hash,
-                            block["block_hash"],
-                            tokens,
-                            lora_id,
+        async for event in events:
+            data = event["data"]
+            if data["type"] == "stored":
+                parent_hash = data["parent_hash"]
+                token_ids = []
+                num_block_tokens = []
+                block_hashes = []
+                for block in data["blocks"]:
+                    token_num_in_block = len(block["tokens"])
+                    block_hash = block["block_hash"]
+                    if token_num_in_block != self._kv_block_size:
```

Comment:
```suggestion
                    if token_num_in_block < self._kv_block_size:
```

Not sure if it will be the case, also log (warning / error?) if framework uses larger block size than expected. I think getting larger block is more critical to proper function of KV routing.

Reply:
will add. good suggestion